compiler/src/diagnostics/graindoc_lexer.re

```diff
@@ -0,0 +1,213 @@
+open Grain_parsing;
+open Graindoc_parser;
+
+let lexbuf_loc = lexbuf => {
+  let (loc_start, loc_end) = Sedlexing.lexing_positions(lexbuf);
+  Location.{loc_start, loc_end, loc_ghost: false};
+};
+
+let ident = [%sedlex.regexp? (xid_start | '_', Star(xid_continue))];
+let semver = [%sedlex.regexp?
+  (
+    Opt('v'),
+    Plus('0' .. '9'),
+    '.',
+    Plus('0' .. '9'),
+    '.',
+    Plus('0' .. '9'),
+    Opt((
+      '-',
+      Star('0' .. '9' | 'A' .. 'z' | '-' | '.'),
+      Opt(('+', Star('0' .. '9' | 'A' .. 'z' | '-'))),
+    )),
+  )
+];
+
+let normalize_semver = semver =>
+  if (semver.[0] == 'v') {
+    Grain_utils.String_utils.slice(
+      ~first=1,
+      ~last=String.length(semver),
+      semver,
+    );
+  } else {
+    semver;
+  };
+
+// HORIZONTAL TABULATION
+// VERTICAL TABULATION
+// SPACE
+// LEFT-TO-RIGHT MARK
+// RIGHT-TO-LEFT MARK
+let blank = [%sedlex.regexp? 0x09 | 0x0B | 0x20 | 0x200E | 0x200F];
```

Comment:
Are some of these defined in `Grain_parsing`? Do we need to redefine them or can we share them?

Reply:
I tried to share them, but it didn't seem possible.